hedyorg/hedy
# 🖊️ Convert input from ask command to booleans

Fixes #5691 
Starting with level 12, we try to convert the input from the ask command to a number (first an int, then a float). If that conversion fails, we keep it as a string. Now that we support boolean values starting with level 15, it makes sense to try and convert the input to a boolean too.

**How to test**
Note that the added automated tests do not cover the full execution of the programs because we cannot fully test the interaction with the 'ask' command. To test this locally, run the following snippet in level 15:
```
a is true
b is ask 'True or False?'
if a == b
    print 'match ' a ' ' b
else
    print 'mismatch ' a ' ' b
```
When the input box appears put in the following: `true`, `false`, `True`, `False`, `15`, `not_a_boolean`. Note that the print statements should output exactly what you typed in, i.e. if you typed `true` you should not get `True`.
Change the language to Bulgarian and paste the following input `вярно` (which is True), `невярно` (which is False).
